src/f32-rdsum/gen/f32-rdsum-7p7x-minmax-rvv-u4v.c

```diff
@@ -0,0 +1,99 @@
+// Auto-generated file. Do not edit!
+//   Template: src/f32-rdsum/rvv.c.in
+//   Generator: tools/xngen
+//
+// Copyright 2024 Imagination Technologies, inc.
+//
+// This source code is licensed under the BSD-style license found in the
+// LICENSE file in the root directory of this source tree.
+
+#include <assert.h>
+
+#include <riscv_vector.h>
+
+#include "xnnpack/common.h"
+#include "xnnpack/reduce.h"
+
+
+void xnn_f32_rdsum_ukernel_7p7x__rvv_u4v(
+    size_t rows,
+    size_t channels,
+    const float* input,
+    size_t input_stride,
+    const float* zero,
+    float* output,
+    const struct xnn_f32_scaleminmax_params params[restrict XNN_MIN_ELEMENTS(1)])
+{
+  assert(rows != 0);
+  assert(channels != 0);
+  assert(input != NULL);
+  assert(output != NULL);
+
+  const float scale = params->scalar.scale;
+  const float min = params->scalar.min;
+  const float max = params->scalar.max;
+
+  size_t input_increment = 7 * input_stride;
+  for (; channels > 0; ) {
+    size_t n = __riscv_vsetvl_e32m4(channels); channels -= n;
+    const float* i0 = input;
+    const float* i1 = (const float*) ((uintptr_t) i0 + input_stride);
+    const float* i2 = (const float*) ((uintptr_t) i1 + input_stride);
+    const float* i3 = (const float*) ((uintptr_t) i2 + input_stride);
+    const float* i4 = (const float*) ((uintptr_t) i3 + input_stride);
+    const float* i5 = (const float*) ((uintptr_t) i4 + input_stride);
+    const float* i6 = (const float*) ((uintptr_t) i5 + input_stride);
+    vfloat32m4_t acc_f32v = __riscv_vfmv_v_f_f32m4(0.f, n);
+
+    for (int r = rows; r > 0; r -= 7) {
+      if XNN_UNPREDICTABLE(r < 2) {
```

Comment:
I still don't like these unpredictable annotations in here. This is very predictable, as we know when we are going to have r < 7. Can this be changed in the code that generates these files?

Reply:
Yes. I will remove the XNN_UNPREDICTABLE.